lib/elemeno.js

```diff
@@ -6,11 +6,22 @@ const request = require('superagent'),
 var apiKey;
 
 function get(path, query, cb) {
+  if(elemeno.caching.enabled) {
+    var cacheValue = elemeno.caching.getValue(path + JSON.stringify(query));
+    if(cacheValue !== null) {
+      // console.log("This comes from the cache.");
+      return cb(cacheValue);
+    }
+  }
+
   request
   .get(baseUrl + path)
   .set('Authorization', apiKey)
   .query(query)
   .end(function(err, res) {
+    if(!err) elemeno.caching.setValue(path + JSON.stringify(query), res.body);
```

Comment:
There will be an error here if you **do not** explicitly enable caching:
```
 this.cache.put(key, data, this.options.refreshTime * 1000);
              ^
TypeError: Cannot read property 'put' of undefined
```

This can be resolved by adding an additional condition, something like:
```
if (elemeno.caching.enabled && !err) elemeno.caching.setValue(path + JSON.stringify(query), res.body);
```